src/lib/hooks/useVersionCheck.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react'
+import { Linking, StyleSheet } from 'react-native'
+import VersionCheck from 'react-native-version-check'
+import { useDialog } from '../undux/utils/dialog'
+
+export default () => {
+  const [showDialog, hideDialog] = useDialog()
+
+  const checkVersion = async () => {
+    const currentVersion = VersionCheck.getCurrentVersion()
+    const latestVersion = await VersionCheck.getLatestVersion()
+
+    const res = await VersionCheck.needUpdate({ currentVersion, latestVersion, depth: 3 })
+
+    if (res.isNeeded) {
+      const onUpdate = () => {
+        Linking.openURL(res.storeUrl)
+      }
+
+      showDialog({
```

Comment:
this should be the same design as the current popup

Reply:
Please apply the sirpy requirments